src/hotspot/share/opto/loopopts.cpp

```diff
@@ -228,6 +228,26 @@ Node* PhaseIdealLoop::split_thru_phi(Node* n, Node* region, int policy) {
     }
   }
 
+  // If the region is a Loop, we are removing the old n,
+  // and need to yank it from the _body. If any phi we
+  // just split through now has no use any more, it also
+  // has to be removed.
+  IdealLoopTree* region_loop = get_loop(region);
+  if (region->is_Loop() && region_loop->_child == nullptr) {
+    region_loop->_body.yank(n);
+    for (uint j = 1; j < n->req(); j++) {
+      Node* in = n->in(j);
+      // Check that in is a phi, and n was its only use.
+      if (in->is_Phi() && in->in(0) == region &&
```

Comment:
Does that work if, say, we're splitting:

`(Add (Phi ..) (Phi ..)`

With a single `Phi` as input twice? Doesn't the `Phi` have 2 uses then (the `Add`, twice)?